### PR TITLE
Added the ability to delete a song from the queue

### PR DESCRIPTION
**Pull request by ykz89**
The !delete command shares the wait time with !next.

### CODE REVIEW

**Review by codeplexnl — APPROVED** [review]

File: Spotify.py [hunk 1]
```diff
@@ -12,80 +12,87 @@
 COMMAND_PLAYLIST = ['playlist', 'pl']
 COMMAND_QUEUE = ['queue', 'q']
 COMMAND_VOLUME_UP = ['volumeup', 'volup', 'vu']
+COMMAND_DELETE = ['delete', 'del', 'd']
 
 USERS = {}
 VOLUME = {}
 
+queueLength = 5
```
Comment (codeplexnl) [inline Spotify.py]: This can be placed in the config

File: Spotify.py [hunk 2]
```diff
@@ -12,80 +12,87 @@
...
-        
+
     @staticmethod
     def handle_message(channel, user, message):
         if config.SLACK_CHANNEL == channel:
             if not message.startswith(COMMAND_PREFIX):
                 return None
-            
+
             command = message[len(COMMAND_PREFIX):]
```
Comment (codeplexnl) [inline Spotify.py]: This will not work if you use params

File: Spotify.py [hunk 3]
```diff
@@ -12,80 +12,87 @@
...
-            
+            numParam = int(filter(str.isdigit, message))
```
Comment (codeplexnl) [inline Spotify.py]: This is only extractable after we know which command we want to execute.

File: Spotify.py [hunk 4]
```diff
@@ -12,80 +12,87 @@
...
+
             if command in COMMAND_QUEUE:
-                queue = Spotify.get_queue(5)
+                queue = Spotify.get_queue(config.QUEUE_LENGTH)
                 numbers = "First {} numbers in the queue".format(len(queue))
+                index = 1
                 for number in queue:
-                    numbers+= "\n{}".format(number)
+                    numbers += "\n{}".format(str(index) + ". " + number)
```
Comment (codeplexnl) [inline Spotify.py]: correctly use the format method

File: Spotify.py [hunk 5]
```diff
@@ -12,80 +12,87 @@
...
+
             command = message[len(COMMAND_PREFIX):]
-            
+            numParam = int(filter(str.isdigit, message))
+
             if command in COMMAND_QUEUE:
-                queue = Spotify.get_queue(5)
+                queue = Spotify.get_queue(config.QUEUE_LENGTH)
                 numbers = "First {} numbers in the queue".format(len(queue))
+                index = 1
```
Comment (codeplexnl) [inline Spotify.py]: To get the index maybe use enumerate

https://stackoverflow.com/questions/522563/accessing-the-index-in-python-for-loops

File: Spotify.py [hunk 6]
```diff
@@ -94,17 +101,27 @@ def handle_message(channel, user, message):
                     return "Volume maxed"
                 else:
                     return "Already used volume up in the last {} minutes. Try again later".format(config.VOLUME_TIME)
-                
+
+            if command in COMMAND_DELETE:
+                if numParam > config.QUEUE_LENGTH or numParam < 1:
+                    return "Please choose a number between 1-" + str(config.QUEUE_LENGTH)
```
Comment (codeplexnl) [inline Spotify.py]: use the format method

File: Spotify.py [hunk 7]
```diff
@@ -94,17 +101,27 @@ def handle_message(channel, user, message):
...
+                elif Spotify.check_user(user):
+                    Spotify.execute_command("del")
```
Comment (codeplexnl) [inline Spotify.py]: del what?

**Review by codeplexnl — CHANGES_REQUESTED** [review]

File: Spotify.py (hunk 1 above)
Comment (ykz89) [reply Spotify.py]: Moved to config

File: Spotify.py (hunk 2 above)
Comment (ykz89) [reply Spotify.py]: Changed to work with params

File: Spotify.py (hunk 3 above)
Comment (ykz89) [reply Spotify.py]: Cahnegd

File: Spotify.py (hunk 5 above)
Comment (ykz89) [reply Spotify.py]: removed index and used enumerate

File: Spotify.py (hunk 4 above)
Comment (ykz89) [reply Spotify.py]: cahnged

File: Spotify.py (hunk 6 above)
Comment (ykz89) [reply Spotify.py]: changed

File: Spotify.py (hunk 7 above)
Comment (ykz89) [reply Spotify.py]: frikandel

File: Spotify.py [hunk 8]
```diff
@@ -94,17 +102,45 @@ def handle_message(channel, user, message):
...
+                        USERS[user] = datetime.datetime.now() + datetime.timedelta(minutes=config.WAIT_TIME)
+                    else:
+                        return "Already skipped or deleted a song from the queue in the last {} minutes. " \
+                               "Try again later".format(config.WAIT_TIME)
+                except ValueError:
+                    return "Please choose a number between 1 - {}".format(str(config.QUEUE_LENGTH))
+
+            if command in COMMAND_RANDOM:
+                param_first = param.partition(' ')[0]
+                param_second = param.partition(' ')[2]
```
Comment (codeplexnl) [inline Spotify.py]: ik denk dat je beter dit kan doen 
 `param_first, param_second = param.split(' ')`

**Review by codeplexnl — CHANGES_REQUESTED** [review]

**Review by codeplexnl — APPROVED** [review]

**Pull request merged**